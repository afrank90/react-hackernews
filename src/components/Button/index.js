import React from 'react';

const Button = ({ onClick, children, className = '' }) => (
  <button onClick={onClick} className={className} type="button">
    {children}
  </button>
);

export default Button;
