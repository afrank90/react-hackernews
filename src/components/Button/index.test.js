import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from './';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
  const props = {
    onClick: () => {},
    className: 'test-class-name',
    children: 'Button text.'
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>More results</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has valid snapshot', () => {
    const component = renderer.create(<Button>More results</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows expected text', () => {
    const element = shallow(<Button {...props} />);

    expect(element.text()).toBe(props.children);
  });

  it('hsa expected class name', () => {
    const element = shallow(<Button {...props} />);

    expect(element.hasClass(props.className)).toBe(true);
  });
});
