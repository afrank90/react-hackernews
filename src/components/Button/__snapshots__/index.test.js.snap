// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has valid snapshot 1`] = `
<button
  className=""
  onClick={undefined}
  type="button"
>
  More results
</button>
`;
